core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/DashboardResource.scala

```diff
@@ -0,0 +1,238 @@
+            key
+          )
+          projectMatchQuery = projectMatchQuery.and(
+            getSearchQuery(true, "project"),
+            key
+          )
+          fileMatchQuery = fileMatchQuery.and(
+            getSearchQuery(true, "file"),
+            key
+          )
+        } else {
+          // When the search query contains multiple words, sub-string search is not supported by MySQL.
+          workflowMatchQuery = workflowMatchQuery.and(
+            getSearchQuery(false, "workflow"),
+            key
+          )
+          projectMatchQuery = projectMatchQuery.and(
+            getSearchQuery(false, "project"),
+            key
+          )
+          fileMatchQuery = fileMatchQuery.and(
+            getSearchQuery(false, "file"),
+            key
+          )
+        }
+      }
+    }
+
+    try {
+
+      // Retrieve workflow resource
+      val workflowQuery =
+        context
+          .select(
+            DSL.inline("workflow").as("resourceType"),
+            WORKFLOW.WID,
+            WORKFLOW.NAME,
+            WORKFLOW.DESCRIPTION,
+            WORKFLOW.CREATION_TIME,
+            DSL.inline("null").as("file_path"),
+            DSL.inline(UInteger.valueOf(0)).as("file_size")
+          )
+          .from(WORKFLOW)
+          .leftJoin(WORKFLOW_USER_ACCESS)
+          .on(WORKFLOW_USER_ACCESS.WID.eq(WORKFLOW.WID))
+          .where(WORKFLOW_USER_ACCESS.UID.eq(user.getUid))
+          .and(
+            workflowMatchQuery
+          )
+
+      // Retrieve project resource
+      val projectQuery = context
+        .select(
+          DSL.inline("project").as("resourceType"),
+          PROJECT.PID,
+          PROJECT.NAME,
+          PROJECT.DESCRIPTION,
+          PROJECT.CREATION_TIME,
+          DSL.inline("null").as("file_path"),
+          DSL.inline(UInteger.valueOf(0)).as("file_size")
+        )
+        .from(PROJECT)
+        .where(PROJECT.OWNER_ID.eq(user.getUid()))
+        .and(
+          projectMatchQuery
+        )
+
+      // Retrieve file resource
+      val fileQuery = context
+        .select(
+          DSL.inline("file").as("resourceType"),
+          FILE.FID,
+          FILE.NAME,
+          FILE.DESCRIPTION,
+          FILE.UPLOAD_TIME.as("creation_time"),
+          FILE.PATH.as("file_path"),
+          FILE.SIZE.as("file_size")
+        )
+        .from(USER_FILE_ACCESS)
+        .join(FILE)
+        .on(USER_FILE_ACCESS.FID.eq(FILE.FID))
+        .join(USER)
+        .on(FILE.OWNER_UID.eq(USER.UID))
+        .where(USER_FILE_ACCESS.UID.eq(user.getUid()))
+        .and(
+          fileMatchQuery
+        )
+      // Retrieve files to which all shared workflows have access
+      val sharedWorkflowFileQuery = context
+        .select(
+          DSL.inline("file").as("resourceType"),
+          FILE.FID,
+          FILE.NAME,
+          FILE.DESCRIPTION,
+          FILE.UPLOAD_TIME.as("creation_time"),
+          FILE.PATH.as("file_path"),
+          FILE.SIZE.as("file_size")
+        )
```

Comment:
We need all of the following info for workflows in order to display in the frontend. Can you copy the select statement from the current `api/workflow/search` and include all the information?
![image](https://github.com/Texera/texera/assets/9412032/61e1d48a-7a31-4cbe-b835-294840eff738)

Reply:
If I include all of these information and since I need to use UNION, here is a sample query that union workflow and project. The query gets super long and hard to read... Do you think if there is a better approach?
```
 SELECT
  'workflow' AS "Type",
  workflow.wid,
  workflow.name,
  workflow.description,
  workflow.creation_time,
  "NULL" as "file_path",
  "NULL" as "file_size",
  -- 9 rows below
  workflow.*,
  workflow_user_access.*,
  --
  null,
  null,
  null,
  null,
  null,
  null
FROM
  workflow
  LEFT JOIN workflow_user_access ON workflow_user_access.wid = workflow.wid
WHERE
  workflow_user_access.uid = 1
union
SELECT
  'project' AS "Type",
  project.pid,
  project.name,
  project.description,
  project.creation_time,
  "NULL" as "file_path",
  "NULL" as "file_size",
  -- workflow attributes 9
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  -- project attributes 6
  project.*
FROM
  project
WHERE
  project.owner_id = 1
  AND (
    MATCH(project.name, project.description) AGAINST ('+project1' IN BOOLEAN mode)
  )
  ```

Reply:
It'd be nice if MySQL allow union by column name, rather than order. Maybe [this](https://stackoverflow.com/questions/3622072/the-used-select-statements-have-a-different-number-of-columns) can help?
If not, I don't have any better ideas